File: sum-of-nodes-with-even-valued-grandparent/sum-of-nodes-with-even-valued-grandparent.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumEvenGrandparent = function(root) {
    if (!root)
        return 0;
    let sum = 0;
    const left = root.left,
          right = root.right;
    if (root.val % 2 === 0){
        sum += left?.left?.val||0;
        sum += left?.right?.val||0;
        sum += right?.left?.val||0;
        sum += right?.right?.val||0;
    }
    sum += left && sumEvenGrandparent(left);
    sum += right && sumEvenGrandparent(right);
    return sum;
};

File: sum-of-nodes-with-even-valued-grandparent/sum-of-nodes-with-even-valued-grandparent.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumEvenGrandparent(self, root: TreeNode) -> int:
        self.total = 0
        
        def dfs(node):
          if node.val % 2 == 0:
            if node.left:
              if node.left.left: self.total += node.left.left.val
              if node.left.right: self.total += node.left.right.val
            if node.right:
              if node.right.left: self.total += node.right.left.val
              if node.right.right: self.total += node.right.right.val
          if node.left: dfs(node.left)
          if node.right: dfs(node.right)
        dfs(root)
        return self.total
            